### PR TITLE
Assert dictPtr offset with FreeOffset when fetching a dictionary

### LINKED ISSUE #461: Dictionary fetching lacks current_ptr assertion
Refer line missing: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/dict.py#L65

Right now the code is only checking if the given dict ptr segment matches. The pointer offset must also match the FreeOffset value of the fetched dictionary.

### PR DESCRIPTION
**Pull request by har777**
Fix #461

### CODE REVIEW

File: pkg/hintrunner/hinter/zero_dict.go [hunk 1]
```diff
@@ -83,36 +83,46 @@ func (dm *ZeroDictionaryManager) NewDefaultDictionary(vm *VM.VirtualMachine, def
...
 // Given a memory address and a key it returns the value held at that position. The address is used
 // to locate the correct dictionary and the key to index on it
 func (dm *ZeroDictionaryManager) At(dictAddr mem.MemoryAddress, key f.Element) (mem.MemoryValue, error) {
-	if dict, ok := dm.dictionaries[dictAddr.SegmentIndex]; ok {
-		return dict.At(key)
+	dict, err := dm.GetDictionary(dictAddr)
+	if err != nil {
+		return mem.UnknownValue, err
+	}
+	value, err := dict.At(key)
```
Comment (TAdev0) [inline pkg/hintrunner/hinter/zero_dict.go:103]: i think just made dict methods private in a previous PR, does it work like this?

Comment (TAdev0) [reply pkg/hintrunner/hinter/zero_dict.go:103]: oh yeah its just conflicts to resolve

Comment (har777) [reply pkg/hintrunner/hinter/zero_dict.go]: fixed conflicts!

**Review by TAdev0 — APPROVED** [review]
lgtm

**Comment by TAdev0** [comment]
@har777  can you resolve conflict?

**Pull request merged**